stix_shifter_modules/vectra/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,678 @@
+        elif expression_01:
+            query = f'{expression_01}'
+        elif expression_02:
+            query = f'{expression_02}'
+        return query
+
+    def _parse_expression(self, expression, qualifier=None) -> str:
+        """
+         Formation of vectra query from ANTLR parsing expression
+        :param expression: expression object, ANTLR parsed expression object
+        :param qualifier: str, default in None
+        :return :None or str
+        """
+        if isinstance(expression, ComparisonExpression):  # Base Case
+            stix_objects = expression.object_path.split(':')
+            mapped_fields_array = self.dmm.map_field(stix_objects[0], stix_objects[1])
+            mapped_field_type = self._get_mapped_field_type(mapped_fields_array)
+            value = self._eval_comparison_value(expression, mapped_field_type)
+            query = self._parse_mapped_fields(value, mapped_fields_array, expression)
+            return query
+
+        elif isinstance(expression, CombinedComparisonExpression):
+            return self._eval_combined_comparison_exp(expression)
+
+        elif isinstance(expression, ObservationExpression):
+            query = self._parse_expression(expression.comparison_expression)
+            return self._add_qualifier(query, qualifier)
+
+        elif hasattr(expression, 'qualifier') and hasattr(expression, 'observation_expression'):
+            if isinstance(expression.observation_expression, CombinedObservationExpression):
+                operator = self._lookup_comparison_operator(expression.observation_expression.operator)
+                expression_01 = self._parse_expression(expression.observation_expression.expr1, expression.qualifier)
+                expression_02 = self._parse_expression(expression.observation_expression.expr2, expression.qualifier)
+                query = f'({expression_01} {operator} {expression_02})'
+            else:
+                query = self._parse_expression(expression.observation_expression, expression.qualifier)
+            if qualifier is not None:
+                query = self._add_qualifier(query, qualifier)
+            return query
+
+        elif isinstance(expression, CombinedObservationExpression):
+            return self._eval_combined_observation_exp(expression, qualifier)
+        elif isinstance(expression, Pattern):
+            return self._parse_expression(expression.expression)
+        else:
+            raise RuntimeError(f'Unknown Recursion Case for expression={expression}, '
+                               f'type(expression)={type(expression)}')
+
+    def parse_expression(self, pattern: Pattern) -> list:
+        """
+         Formation of vectra query from ANTLR parsing expression.
+        :param pattern: expression object, ANTLR parsed expression object
+        """
+        query = self._parse_expression(pattern)
+        if self.options['result_limit'] > DEFAULT_LIMIT:
```

Comment:
is there a limitations on the result limit? not sure why you are replacing result_limit to  `DEFAULT_LIMIT`

Reply:
removed the default limit; it's not required.